ResearchKit/Common/ORKTintedImageView.m

```diff
@@ -77,17 +158,25 @@ - (UIImage *)imageByTintingImage:(UIImage *)image {
     
     UIColor *tintColor = self.tintColor;
     CGFloat screenScale = self.window.screen.scale; // Use screen.scale; self.contentScaleFactor remains 1.0 until later
-    if ((![_appliedTintColor isEqual:tintColor] || !ORKCGFloatNearlyEqualToFloat(_appliedScaleFactor, screenScale))) {
+    if (screenScale > 0 && (![_appliedTintColor isEqual:tintColor] || !ORKCGFloatNearlyEqualToFloat(_appliedScaleFactor, screenScale))) {
         _appliedTintColor = tintColor;
         _appliedScaleFactor = screenScale;
         
         if (!ORKIsImageAnimated(image)) {
-            _tintedImage = [image imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+            if (_enableTintedImageCaching) {
+                _tintedImage = [[ORKTintedImageCache sharedCache] tintedImageForImage:image tintColor:tintColor scale:screenScale];
+            } else {
+                _tintedImage = [image imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+            }
         } else {
             // Manually apply the tint for animated images (template rendering mode doesn't work: <rdar://problem/19792197>)
             NSMutableArray *images = [NSMutableArray array];
             for (UIImage *image in image.images) {
-                [images addObject:ORKImageByTintingImage(image, tintColor, screenScale)];
+                if (_enableTintedImageCaching) {
+                    [[ORKTintedImageCache sharedCache] tintedImageForImage:image tintColor:tintColor scale:screenScale];
```

Comment:
No, it adds each tinted frame image from the animated image. If you build the tinted animated image a second time, it should retrieve the individual frames from the cache.

But yeah, the fact that loop variable has the same name as the argument it's confusing, I'm going to rename it.